### PR TITLE
Placed error on dir in the proper place

### PR DESCRIPTION
**Pull request by cpviolator**

### CODE REVIEW

File: include/kernels/laplace.cuh [hunk 1]
```diff
@@ -159,7 +161,8 @@ namespace quda
                                                                          active);
       break;
     default:
-      errorQuda("Unsupported laplace direction %d given", arg.dir);
+      applyLaplace<Float, nDim, nColor, nParity, dagger, kernel_type, -1>(out, arg, coord, x_cb, parity, idx,
```
Comment (mathiaswagner) [inline include/kernels/laplace.cuh]: Can we just do something like:
```C++
case 3:
  apply < ...., 3>... 
  break;
case 4:
default:
  apply <...., -1>
```

here?

Comment (cpviolator) [reply include/kernels/laplace.cuh]: Yes, that's much neater.

**Comment by maddyscientist** [comment]
Looks like the switch on `arg.dir` is a run-time check not a compile time check.  This should be changed to a compile time switch, e.g., make `dir` a template parameter rather than a member of the argument struct.

**Comment by cpviolator** [comment]
@maddyscientist Similar to the compile time switch on dir for covDev?

https://github.com/lattice/quda/blob/develop/lib/covDev.cu#L340

**Comment by maddyscientist** [comment]
Yeh, it probably doesn't matter that much.  I'll just merge this as is, and if we ever need 1% more performance out of Laplace, we'll know where to get it 😉

**Pull request merged**